pgmpy/factors/Factor.py

```diff
@@ -191,39 +191,26 @@ def marginalize(self, variables, inplace=True):
             factor = self
         else:
             factor = Factor(self.scope(), self.cardinality, self.values)
-
+        marginalize_index = np.array(np.where(np.in1d(factor.scope(),variables)))
+        assign = np.array(factor.cardinality)
+        assign[marginalize_index] = -1
+        sum_variable = np.zeros(len(factor.cardinality))
+        sum_variable[marginalize_index] = -1
+        marg_factor = []
+        for i in itertools.product(*[range(index) for index in factor.cardinality[np.where(assign!=-1)[0]]]):
```

Comment:
What not simply use `assign[assign != -1]` instead of `factor.cardinality[np.where(assign != -1)[0]]` ?